lib/pos_mempool.go

```diff
@@ -0,0 +1,446 @@
+package lib
+
+import (
+	"fmt"
+	"github.com/emirpasic/gods/sets/treeset"
+	"github.com/golang/glog"
+	"math/big"
+)
+
+// ========================
+//	TransactionRegister
+// ========================
+
+// TransactionRegister is the primary transaction store of the PoS Mempool. The register is responsible for determining
+// the Fee-Time ordering of transactions. The operations supported by the register are: adding a transaction, removing
+// a transaction, iterating through all transactions in fee-time order, and retrieving all transactions ordered in
+// fee-time. The TransactionRegister doesn't perform any validation on the transactions, it just accepts the provided
+// MempoolTx and adds it to the appropriate FeeBucket and TimeBucket.
+type TransactionRegister struct {
+	// buckets is the data structure storing the transactions in fee-time order.
+	buckets *FeeBucket
+	// txnMembership is a map of transaction hashes. It is used to determine existence of a transaction in the register.
+	txnMembership map[BlockHash]struct{}
+	// totalTxnSize is the total size of all transactions in the register.
+	totalTxnSize uint64
+
+	params       *DeSoParams
+	globalParams *GlobalParamsEntry
```

Comment:
We store the full globalParams here, but we really only need to store the the base fee rate and multiplier. IMO, you can extract those two fields, compute their big.Float values, and store them as separate fields here instead.

That way, if the GlobalParamsEntry is mutated externally, it won't break the fee bucket computation.

Reply:
I like!